File: SlidingWindowandTwoPointers/Longest Festival Streak/manasvi9745.py
def longest_festival_streak(arr, k):
    left = 0
    s = 0
    max_len = 0

    for right in range(len(arr)):
        s += arr[right]

        while s > k:
            s -= arr[left]
            left += 1

        max_len = max(max_len, right - left + 1)

    return max_len

File: SlidingWindowandTwoPointers/Longest Festival Streak/manasvi9745.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        long k = sc.nextLong();
        long[] arr = new long[n];

        for (int i = 0; i < n; i++)
            arr[i] = sc.nextLong();

        int left = 0;
        long sum = 0;
        int maxLength = 0;

        for (int right = 0; right < n; right++) {
            sum += arr[right];

            while (sum > k) {
                sum -= arr[left];
                left++;
            }

            maxLength = Math.max(maxLength, right - left + 1);
        }

        System.out.println(maxLength);
    }
}